src/librustc_mir/interpret/place.rs

```diff
@@ -583,35 +583,38 @@ where
         use rustc::mir::Place::*;
         use rustc::mir::PlaceBase;
         Ok(match *mir_place {
-            Base(PlaceBase::Promoted(ref promoted)) => {
```

Comment:
Maybe keep the original code, but directly destructure this arm via `Base(PlaceBase::Static(Static { promoted: Some(promoted), ty, .. }))` and the other arm via `Base(PlaceBase::Static(Static { promoted: None, ty, def_id }))`